### ISSUE TITLE
Day view FILL alignment strategy doesn't work with intersecting entries

### ISSUE THREAD

**Issue by ftrela** [dlemmermann/CalendarFX#86]
Inside the entry view factory the Entry1 entry view has set specific pref width and alignment strategy to `ALIGN_LEFT`:
```
entryView.setPrefWidth(20);
entryView.setAlignmentStrategy(EntryViewBase.AlignmentStrategy.ALIGN_LEFT);
```
The Entry2 and Entry3 entries have default alignment strategy `EntryViewBase.AlignmentStrategy.FILL`.

When the Entry1, Entry2 and Entry3 intersect and the Entry1 start time is earlier than the Entry2 and Entry3 start times, then the Entry2 and Entry3 do not fill the rest of the day view. There is a gap between Entry1 and Entry2. Please look at the image below:
![CalendarFX_fill_strategy](https://user-images.githubusercontent.com/20297211/79433952-0c070a80-7fce-11ea-9bf5-e9c23d4315f2.png)

I would expect that entries with the `EntryViewBase.AlignmentStrategy.FILL` strategy will fill all possible space of day view, respecting pref width of intersecting entries with alignment strategies different from `EntryViewBase.AlignmentStrategy.FILL` strategy.

What do you think? Is it possible?

**Comment by dlemmermann** [comment]
I have added the property autoLayout to DayViewBase, which means it is also available for the ResourceCalendarView. There the value is set to false, so CalendarFX will no longer try to place entries with overlapping time intervals in separate columns. Please let me know if this solves this issue.

**Comment by ftrela** [comment]
I have seen you removed `autoLayout` property and added the `OverlapResolutionStrategy` instead. I understand that for achieving my goal I would have to use the `OverlapResolutionStrategy.OFF` right? But still it is not quite clear to me how should I get the information about how many operations are overlapping with the one for which I'm at the moment calculating a width and a position (in the scenario when I would have to determine their position by myself). Because the behavior I would expect is:
![calendarfx_entry_width](https://user-images.githubusercontent.com/20297211/81182357-f58d1700-8fad-11ea-9f64-505fd46e034f.png)
So at the end I would like to respect the overlapping information but calculate the width of overlapping entries not just basing on the number of entries overlapping with each other but also their width if it has been changed by the developer. So in that case the default `OverlapResolutionStrategy.TIME_BOUNDS` looks to me like the most appealing one.
To sum up, wouldn't it be easier if the overlapping rendering algorithm would consider the information about the width of an entry when it has been directly set using `entryView.setPrefWidth(10)`? Maybe the `entryView` could have something like `entryView.setFixedWidth(10)` so that the algorithm would be sure that it has been done intentionally. What do you think?

**Comment by dlemmermann** [comment]
We are now at a point where I would suggest that you take a look at the "resolver" classes yourself and come up with a layout algorithm that you think would work for you. When it comes to automatic layout there are many many cases to consider and really only when you try to implement this yourself you will notice where the problems are.

The current resolver detects "clusters" of entries and then creates enough "columns" inside the cluster to make sure no entries overlap. In your example above a fourth entry could fit "below" entry 1 but because its pref width is not set it would not fit into the first column of this cluster. So maybe we can put it into column 2 but maybe its pref width is too big for column 2, so now it needs to span 2 columns? Column spanning is not part of the current solution at all. Working on an algorithm that works for all your situations could easily take a week or more to find. If somebody does it who is not on the project that requires it and with whom you can only communicate via issue tickets, that might take a month. I would strongly suggest you try to come up with a resolver yourself by taking the code that is there and trying to see if you can come up with something.

**Comment by dlemmermann** [comment]
Also, I need to know if you are working for PSI or not. Hard to tell from the handle.

**Comment by ftrela** [comment]
Yes. I'm working for PSI. I know it's a very complicated and time consuming problem and that's why I came to you with this :)
Anyway, as you already know, we came up with another solution for this problem.
We add a narrow column (resource) to the left of each resource that we have and fill it with our narrow entries.
Thanks to that we have narrow entries always on the left and quite independent from entries that we have in the main resource.
To change the narrow column width, we manipulate the `ColumnConstraints` of the `ResourceCalendarViewSkin.HeaderGridPane` and `ResourceCalendarViewSkin.BodyGridPane`.
The only problem was with getting the `HeaderGridPane` and `BodyGridPane` instances from the `ResourceCalendarView` object, but we have used the `javafx.scene.Parent#lookup` and `javafx.scene.layout.Pane#getChildren` methods and type casting:
```
GridPane gridPane = (GridPane)lookup( ".resource-calendar-container" );
ResourceCalendarViewSkin.HeaderGridPane headerGridPane =
	(ResourceCalendarViewSkin.HeaderGridPane)gridPane.getChildren().get( 2 );
ResourceCalendarViewSkin.BodyGridPane bodyGridPane =
	(ResourceCalendarViewSkin.BodyGridPane)gridPane.getChildren().get( 3 );
```
And it looks like this:
![resource_calendar_view_with_narrow_entries](https://user-images.githubusercontent.com/20297211/82026210-b8eaab00-9692-11ea-8abe-52af2f43159b.png)


So I'm closing this issue and thank you for your time.
